diff --git a/i18n/qgis_ru.ts b/i18n/qgis_ru.ts
index efe34d6..1815d94 100644
--- a/i18n/qgis_ru.ts
+++ b/i18n/qgis_ru.ts
@@ -76651,4 +76651,11 @@ Convert a string into Qt data type.
         <translation>...</translation>
     </message>
 </context>
+<context>
+    <name>QgisApp</name>
+    <message>
+        <source>GIS kontrolya stroitel'stva VOLS</source>
+        <translation>ГИС контроля строительства ВОЛС</translation>
+    </message>
+</context>
 </TS>
diff --git a/src/app/qgisapp.cpp b/src/app/qgisapp.cpp
index 37aa956..269412b 100644
--- a/src/app/qgisapp.cpp
+++ b/src/app/qgisapp.cpp
@@ -340,6 +340,7 @@ const int AFTER_RECENT_PATHS = 321;
   */
 static void setTitleBarText_( QWidget & qgisApp )
 {
+  /*
   QString caption = QgisApp::tr( "QGIS " );
 
   if ( QString( QGis::QGIS_VERSION ).endsWith( "Master" ) )
@@ -368,7 +369,8 @@ static void setTitleBarText_( QWidget & qgisApp )
   {
     caption += " - " + QgsProject::instance()->title();
   }
-
+  */
+  QString caption = QgisApp::tr( "GIS kontrolya stroitel'stva VOLS" );
   qgisApp.setWindowTitle( caption );
 } // setTitleBarText_( QWidget * qgisApp )
 
@@ -703,7 +705,10 @@ QgisApp::QgisApp( QSplashScreen *splash, bool restorePlugins, QWidget * parent,
   connect( QgsMapLayerActionRegistry::instance(), SIGNAL( changed() ), this, SLOT( refreshActionFeatureAction() ) );
 
   // set application's caption
+  /*
   QString caption = tr( "QGIS - %1 ('%2')" ).arg( QGis::QGIS_VERSION ).arg( QGis::QGIS_RELEASE_NAME );
+  */
+  QString caption = QgisApp::tr( "GIS kontrolya stroitel'stva VOLS" );
   setWindowTitle( caption );
 
   QgsMessageLog::logMessage( tr( "QGIS starting..." ), QString::null, QgsMessageLog::INFO );
diff --git a/src/plugins/CMakeLists.txt b/src/plugins/CMakeLists.txt
index 6602235..dd120f6 100644
--- a/src/plugins/CMakeLists.txt
+++ b/src/plugins/CMakeLists.txt
@@ -6,7 +6,7 @@ ADD_SUBDIRECTORY(interpolation)
 ADD_SUBDIRECTORY(oracle_raster)
 ADD_SUBDIRECTORY(raster_terrain_analysis)
 ADD_SUBDIRECTORY(coordinate_capture)
-ADD_SUBDIRECTORY(dxf2shp_converter)
+#ADD_SUBDIRECTORY(dxf2shp_converter)
 ADD_SUBDIRECTORY(evis)
 ADD_SUBDIRECTORY(spatialquery)
 ADD_SUBDIRECTORY(roadgraph)
